src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -1,78 +1,109 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
 import re
 import time
 import sys
+import os
+import tarfile
+import requests
 from datetime import datetime, timedelta
 from io import BytesIO
 import time
+import tempfile
 import colorama
+import pytz
 from azure.common import AzureHttpError
-from azure.storage.blob import AppendBlobService, ContainerPermissions
+from azure.storage.blob import (
+    BlockBlobService,
+    AppendBlobService,
+    ContainerPermissions
+)
 from .sdk.models import (
     QueueBuildRequest,
     DockerBuildParameters,
     PlatformProperties,
     Build,
-    BuildArgument
+    BuildArgument,
+    SourceUploadDefinition
 )
-from knack.util import CLIError
 from ._utils import (
-    get_resource_group_name_by_registry_name,
+    get_resource_group_name_by_registry_name
 )
-from azure.cli.core.commands import LongRunningOperation
-import pytz
+from azure.cli.core.commands import (
+    LongRunningOperation
+)
+from knack.util import CLIError
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+class BlobInfo(object):
```

Comment:
The class seems to be an overkill since each field is passed separately.

Reply:
Not sure if I understand. The class takes a full sas url and break it down to account/container/endpoint/sas.

Reply:
You can have a method that returns all four of them. The reason is that I didn't think the object is passed around anywhere so seems to be an overkill. The current approach is fine too.

Reply:
updated. thanks